concordium-consensus/src/Concordium/KonsensusV1/Consensus/Blocks.hs

```diff
@@ -1229,6 +1272,99 @@ bakeBlock BakeBlockInputs{..} = do
+                ( Hash.hashOfHashes
+                    (theFinalizationCommitteeHash shiCurrentFinalizationCommitteeHash)
+                    (theFinalizationCommitteeHash shiNextFinalizationCommitteeHash)
+                )
+            )
+  where
+    blockHeightInfoHash BlockHeightInfo{..} = Hash.hash $ Serialize.runPut $ do
+        Serialize.put bhiAbsoluteBlockHeight
+        Serialize.put bhiGenesisIndex
+        Serialize.put bhiRelativeBlockHeight
+
+-- | Extract information from SkovData and the block state to compute the result block hash.
+computeBlockResultHash ::
+    forall m.
+    ( MonadState (SkovData (MPV m)) m,
+      BlockStateStorage m,
+      BlockState m ~ HashedPersistentBlockState (MPV m),
+      IsConsensusV1 (MPV m)
+    ) =>
+    -- | The block state right after executing the block.
+    HashedPersistentBlockState (MPV m) ->
+    -- | The relative block height for the block.
+    BlockHeight ->
+    m BlockResultHash
+computeBlockResultHash newState relativeBlockHeight = do
+    theBlockStateHash <- getStateHash newState
+    transactionOutcomesHash <- getTransactionOutcomesHash newState
+    currentFinalizationCommitteeHash <- use $ to bakersForCurrentEpoch . bfFinalizerHash
+    nextFinalizationCommitteeHash <- do
+        -- Attempt to get the finalization committee from SkovData otherwise compute it from the
+        -- information in the block state.
+        currentEpoch <- use (roundStatus . rsCurrentEpoch)
+        nextSkovBakersAndFinalizers <- gets (getBakersForEpoch (currentEpoch + 1))
+        case nextSkovBakersAndFinalizers of
+            Just bakersAndFinalizers -> return $ bakersAndFinalizers ^. bfFinalizerHash
+            Nothing -> do
+                nextFullBakers <- getNextEpochBakers newState
+                nextFinalizationParameters <- getNextEpochFinalizationCommitteeParameters newState
+                let nextFinalizationCommittee = computeFinalizationCommittee nextFullBakers nextFinalizationParameters
+                return $ computeFinalizationCommitteeHash nextFinalizationCommittee
+    blockHeightInfo <- do
+        genesisBlockHeightInfo <- use genesisBlockHeight
+        return
+            BlockHeightInfo
+                { bhiAbsoluteBlockHeight = localToAbsoluteBlockHeight (gbhiAbsoluteHeight genesisBlockHeightInfo) relativeBlockHeight,
+                  bhiGenesisIndex = gbhiGenesisIndex genesisBlockHeightInfo,
+                  bhiRelativeBlockHeight = relativeBlockHeight
+                }
+    return $
+        makeBlockResultHash
+            BlockResultHashInput
+                { shiBlockStateHash = theBlockStateHash,
+                  shiTransationOutcomesHash = transactionOutcomesHash,
+                  shiCurrentFinalizationCommitteeHash = currentFinalizationCommitteeHash,
+                  shiNextFinalizationCommitteeHash = nextFinalizationCommitteeHash,
+                  shiBlockHeightInfo = blockHeightInfo
+                }
+
```

Comment:
I moved the types and the `makeBlockResultHash` function, but not the `computeBlockResultHash` since it relies on context, which is only relevant for this file.